be/src/exprs/runtime_filter.cpp

```diff
@@ -439,9 +481,20 @@ class RuntimePredicateWrapper {
     }
 
     Status merge(const RuntimePredicateWrapper* wrapper) {
-        DCHECK(_filter_type == wrapper->_filter_type);
-        if (_filter_type != wrapper->_filter_type) {
-            return Status::InvalidArgument("invalid filter type");
+        bool can_not_merge_in_or_bloom
+            = _filter_type == RuntimeFilterType::IN_OR_BLOOM_FILTER &&
+                  (wrapper->_filter_type != RuntimeFilterType::IN_FILTER
+                   && wrapper->_filter_type != RuntimeFilterType::BLOOM_FILTER);
+
+        bool can_not_merge_other = _filter_type != RuntimeFilterType::IN_OR_BLOOM_FILTER
+                               && _filter_type != wrapper->_filter_type;
+
+        if (can_not_merge_in_or_bloom || can_not_merge_other) {
+            std::stringstream msg;
```

Comment:
Do not using `stringstream` here.
Same as others

Reply:
already change to CHECK